pkg/controller/directvolumemigration/rsync.go

```diff
@@ -352,12 +391,10 @@ func (t *Task) createRsyncTransferPods() error {
 
 	// Generate pubkey bytes
 	// TODO: Use a secret for this so we aren't regenerating every time
-	publicRsaKey, err := ssh.NewPublicKey(t.SSHKeys.PublicKey)
+	pubKeyBytes, err := t.getPubKeyBytes()
```

Comment:
is there any plan to mount the secret directly instead of populating `pubKeyBytes` as an environment [variable](https://github.com/konveyor/mig-controller/pull/828/files#diff-7299a3c4e600e2733571d1d39c64bbfe6c7e448a6bb15aa472514a173d3095c0R515)

The problem I see is:
1. Assume a dvm run failed in a way that would leave the pods but delete the secret
2. The pod yaml has data from old secret but the controller creates a new secret in the next run.
3. The controller will not patch the pod yaml with this new secret data.


Had a secret been mounted on the pod directly, if the secret is patched the pods will see the results in around 30 seconds max (I think)

I also think the above mentioned failure scenario is very rare to occur, but just calling out the thats I had that could fail this

Reply:
Yeah interesting point... that is certainly possible as a corner case.

For this to work we'd need to create the secrets in each namespace which comes with it's own set of possible corner cases if one namespace has the secret and the other doesn't between the two clusters.

I think we should perhaps open an issue to track this corner case but moving it to one secret per namespace also scares me so I'd like to think further on it.